Firestore/Source/Local/FSTLevelDBMigrations.mm

```diff
@@ -61,30 +63,39 @@ static void SaveVersion(FSTLevelDBSchemaVersion version, LevelDbTransaction *tra
   transaction->Put(key, version_string);
 }
 
-/**
- * This function counts the number of targets that currently exist in the given db. It
- * then reads the target global row, adds the count to the metadata from that row, and writes
- * the metadata back.
- *
- * It assumes the metadata has already been written and is able to be read in this transaction.
- */
-static void AddTargetCount(LevelDbTransaction *transaction) {
-  auto it = transaction->NewIterator();
-  std::string start_key = [FSTLevelDBTargetKey keyPrefix];
-  it->Seek(start_key);
-
-  int32_t count = 0;
-  while (it->Valid() && absl::StartsWith(it->key(), start_key)) {
-    count++;
-    it->Next();
+static void DeleteEverythingWithPrefix(const std::string &prefix, leveldb::DB *db) {
+  bool more_deletes = true;
+  while (more_deletes) {
+    LevelDbTransaction transaction(db, "Delete everything with prefix");
+    auto it = transaction.NewIterator();
+
+    more_deletes = false;
+    for (it->Seek(prefix); it->Valid() && absl::StartsWith(it->key(), prefix); it->Next()) {
+      if (transaction.changed_keys() >= 1000) {
+        more_deletes = true;
+        break;
+      }
+      transaction.Delete(it->key());
+    }
+
+    transaction.Commit();
```

Comment:
FWIW, I think we could end up committing an empty transactions if the number of keys is a multiple of 1000, but this must be safe since your test should end up in this case (I think?).  So I don't think any action is necessary.

Reply:
Empty commits work fine. I'm not sure it's worth the code to avoid them.